### ISSUE TITLE
promise-function-async false positives when returning Buffer

### ISSUE THREAD

**Issue by hayes** [typescript-eslint/typescript-eslint#805]
**Repro**
```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/promise-function-async": "error",

  }
}
```

```TS
function example(): Buffer {
  return Buffer.from('foo');
}
```

**Expected Result**

No error

**Actual Result**
`Functions that return promises must be async.`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.13.0` |
| `npm`                              | `6.0.0` |

**Comment by bradzacher** [comment]
use the option `allowAny: true`.
This has been changed to be the default in v2

**Comment by hayes** [comment]
ah, I see.  This is because this is because Buffer is a global type rather than something defined locally or imported.
